lib/matplotlib/axis.py

```diff
@@ -2584,7 +2605,21 @@ def _update_label_position(self, renderer):
             self.label.set_position(
                 (left - self.labelpad * self.figure.dpi / 72, y)
             )
+        elif self.label_position == 'axesleft':
+            spinebbox = self.axes.bbox
+            bbox = mtransforms.Bbox.union(bboxes + [spinebbox])
```

Comment:
Do you have a test of this if the spines are not floating?

Reply:
I've added a test now for normal axes; and thank you - because in doing so I did discover a little bug. Now the new keywords behave as a superset of the usual `top`, `left`, etc.